source/extensions/omni.isaac.lab/test/assets/test_rigid_object.py

```diff
@@ -79,6 +79,58 @@ def generate_cubes_scene(
     return cube_object, origins
 
 
+def generate_heterogeneous_cubes_scene(height=1.0, device: str = "cuda:0") -> tuple[RigidObject, torch.Tensor]:
+    """Generate a scene with the provided number of cubes.
+
+    Args:
+        height: Height of the cubes.
+        device: Device to use for the simulation.
+
+    Returns:
+        A tuple containing the rigid object representing the cubes and the origins of the cubes.
+
+    """
+    num_cubes = 2
+    origins = torch.tensor([(i * 1.0, 0, height) for i in range(num_cubes)]).to(device)
+    # Create Top-level Xforms, one for each cube
+    for i, origin in enumerate(origins):
+        prim_utils.create_prim(f"/World/Table_{i}", "Xform", translation=origin)
+
+    # Resolve spawn configuration
+    spawn_cfg_1 = sim_utils.UsdFileCfg(
+        usd_path=f"{ISAAC_NUCLEUS_DIR}/Props/Blocks/block_instanceable.usd",
+        rigid_props=sim_utils.RigidBodyPropertiesCfg(),
+    )
+    spawn_cfg_2 = sim_utils.UsdFileCfg(
+        usd_path=f"{ISAAC_NUCLEUS_DIR}/Props/Blocks/basic_block.usd",
+        rigid_props=sim_utils.RigidBodyPropertiesCfg(),
+    )
+
+    # Create rigid object
+    cube_object_cfg_1 = RigidObjectCfg(
+        prim_path="/World/Table_0/Object",
+        spawn=spawn_cfg_1,
+        init_state=RigidObjectCfg.InitialStateCfg(pos=(0.0, 0.0, height)),
+    )
+    cube_object_cfg_2 = RigidObjectCfg(
+        prim_path="/World/Table_1/Object",
+        spawn=spawn_cfg_2,
+        init_state=RigidObjectCfg.InitialStateCfg(pos=(0.0, 0.0, height)),
+    )
+    cube_object_cfg = RigidObjectCfg(
+        prim_path="/World/Table_.*/Object",
+        spawn=None,
+        init_state=RigidObjectCfg.InitialStateCfg(pos=(0.0, 0.0, height)),
+        use_first_matching_path=False,
```

Comment:
I am a bit confused. Why is this needed? I thought this case is automatically handled. Otherwise `/World/envs/env_.*/Object` would have never worked.